addons/l10n_in/views/res_country_state_view.xml

```diff
@@ -12,4 +12,15 @@
         </field>
     </record>
 
+    <record id="l10n_in_view_country_state_tree_inherit" model="ir.ui.view">
+        <field name="name">l10n.in.res.country.state.tree.inhert</field>
+        <field name="model">res.country.state</field>
+        <field name="inherit_id" ref="base.view_country_state_tree"/>
+        <field name="arch" type="xml">
+            <field name="code" position="after">
+                <field name="l10n_in_tin" readonly="l10n_in_tin"/>
```

Comment:
```suggestion
                <field name="l10n_in_tin" readonly="l10n_in_tin" optional="hide"/>
```
@sham-odoo @ajjo-odoo 

I would suggest to make this field optional hide because we don't want other than Indian companies to see this field

cc- @jco-odoo

Reply:
Be careful with the readonly however.  Normally, it should be only visible in the form view anyways.

Reply:
> Be careful with the readonly however. Normally, it should be only visible in the form view anyways.

Hello JCO
thanks for the idea, when i observed to find out the form view, i could not find out the view, 
secondly we may allow editable at following screen in country with TIN column. and readonly over a fed state list view
![image](https://github.com/odoo/odoo/assets/93132516/a9cd6319-3528-4036-95e0-fc27a5b6e97c)

Reply:
> Be careful with the readonly however. Normally, it should be only visible in the form view anyways.

Hello JCO
just updated mockup
https://link.excalidraw.com/l/65VNwvy7c4X/2EUqZNtreWy

TIN field - view only in debug mode and kept it editable.